### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by ThibaultKlein**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -659,4 +660,13 @@ internal final class SubClass: BaseClass {
...
 *Rationale* Subclassing in instances where the original class was not built to support subclasses can be a common source of bugs. Marking classes as `final`
-indicates that it was developed under the assumption that it would act on its own without regard for subclasses. 
+indicates that it was developed under the assumption that it would act on its own without regard for subclasses.
+
+### Documentation ###
+
+Well documented code is critical to help other developers understand what the code is doing. Every **public** and **internal** types should be documented.
+Additionally developers should annotate a private function if the behavior of it is not trivial using the regular comment syntax `//`.
+
+[See also](BestPractices.md) section about documentation.
```
Comment (paulmiard) [inline README.md]: See also some of [our preferred practices] when documenting code(BestPractices.md#documentation).

File: README.md [hunk 2]
```diff
@@ -659,4 +660,13 @@ internal final class SubClass: BaseClass {
 
 
 *Rationale* Subclassing in instances where the original class was not built to support subclasses can be a common source of bugs. Marking classes as `final`
-indicates that it was developed under the assumption that it would act on its own without regard for subclasses. 
+indicates that it was developed under the assumption that it would act on its own without regard for subclasses.
+
+### Documentation ###
+
+Well documented code is critical to help other developers understand what the code is doing. Every **public** and **internal** types should be documented.
+Additionally developers should annotate a private function if the behavior of it is not trivial using the regular comment syntax `//`.
```
Comment (paulmiard) [inline README.md]: `a private function` -> `a piece of code` ??

Comment (paulmiard) [inline README.md]: `the behavior of it` -> `its behavior`

Comment (ThibaultKlein) [inline README.md]: I want to explain that even private code should be commented if the developer thinks the implementation is not trivial, I can change to `private piece of code` if you think `function` is too specific.

I updated for `its behavior`

File: README.md (hunk 1 above)
Comment (ThibaultKlein) [inline README.md]: What about `See our [best practices](BestPractices.md#documentation) about documenting the code.`

Comment (paulmiard) [inline README.md]: Yeah. I'd say the word `also` is good but it's good that way as well

File: README.md (hunk 2 above)
Comment (paulmiard) [inline README.md]: I'd say any code including private then. Both `private` and `function` is too specific

Comment (ThibaultKlein) [inline README.md]: Sounds good to me, I made the update

**Pull request merged**